### PR TITLE
Add prettier, eslint, husky and lintstaged

### PR DESCRIPTION
**Pull request by Johnnycus**
ESLint suggests replacing `for...in` with array loops so I added `// TODO` comments there. Or we can turn that rule off.

### CODE REVIEW

**Comment by billythedummy** [comment]
> ESLint suggests replacing `for...in` with array loops so I added `// TODO` comments there. Or we can turn that rule off.

Lets keep that rule, there seems to be only 1 instance where `for (... in ...)` is used and thats in `reverse()` in `stake-pool/utils.ts`. Could you refactor that to iterate `Object.keys()` instead? Thanks!

**Comment by Johnnycus** [comment]
> > ESLint suggests replacing `for...in` with array loops so I added `// TODO` comments there. Or we can turn that rule off.
> 
> Lets keep that rule, there seems to be only 1 instance where `for (... in ...)` is used and thats in `reverse()` in `stake-pool/utils.ts`. Could you refactor that to iterate `Object.keys()` instead? Thanks!

Done, there were also some instances of `for...of` which I also replaced with `forEach`. I only kept one `for...of` that had `async/await`. We can replace it with `map`, but this will run the, in parallel

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "no-restricted-imports": [
+      "error",
+      {
+        "patterns": [".*"]
+      }
+    ],
+    "@typescript-eslint/no-use-before-define": "warn",
+    "no-plusplus": "off",
+    "import/no-cycle": "warn",
+    "no-await-in-loop": "off",
```
Comment (billythedummy) [inline .eslintrc.json:31]: think we can keep this on and just use eslint disable comments when we need to break the rule like what you did in signAndSendTransactionSequence?

**Review by billythedummy — APPROVED** [review]
there's gonna be a lot to add once https://github.com/igneous-labs/stake-pool-sdk/pull/36 is merged, sorry about that

File: .eslintrc.json (hunk 1 above)
Comment (Johnnycus) [reply .eslintrc.json:31]: Agreed, will update

**Pull request merged**